Protecs/src/test/java/uk/ac/ucl/protecs/sim/WorkplaceTesting.java

```diff
@@ -0,0 +1,220 @@
+	private final String params;
+	
+	public WorkplaceTesting(String fileName) {
+		this.params = fileName;
+	}
+
+	@Test
+	public void checkPopulationWorkplacesAreBeingLoaded() {
+		// workplaces are read in from the csv file now, initialise the simulation
+		WorldBankCovid19Sim sim = helperFunctions.CreateDummySim("src/main/resources/workplace_bubbles_params.txt");
+		sim.start();
+		// check that the workplace locations have been created
+		Assert.assertTrue(sim.workplaces.size() > 0);
+
+	}
+	@Test
+	public void checkWorkplaceBubblesAreBeingMade() {
+		// bubbles are created after during the loading in process of the population
+		WorldBankCovid19Sim sim = helperFunctions.CreateDummySim("src/main/resources/workplace_bubbles_params.txt");
+		sim.start();
+		// check that everyone has a bubble associated with their workplace
+		boolean hasBeenGivenABubble = true;
+		for (Person p: sim.agents) {
+			if (p.getWorkBubble().isEmpty()) {hasBeenGivenABubble = false;}
+		}
+		Assert.assertTrue(hasBeenGivenABubble);
+	}
+	@Test
+	public void checkWorkplaceBubblesContainEveryoneInWorkplace() {
+		// bubbles are created after during the loading in process of the population
+
+		WorldBankCovid19Sim sim = helperFunctions.CreateDummySim("src/main/resources/workplace_bubbles_params.txt");
+		sim.start();
+		// create a function to search through the population and get people belonging to certain bubbles
+		Map<String, List<Person>> belongingToBubble = sim.agents.stream().collect(
+				Collectors.groupingBy(
+						Person::checkWorkplaceID
+						)
+				);
+		boolean bubbleContainsEveryoneInWorkplace = true;
+		// check that for each person, everyone who belongs to the workplace features in their bubble
+		for (Workplace w: sim.workplaces) {
+			String workplaceID = w.getId();
+			List<Person> thisBubble = belongingToBubble.get(workplaceID);
+			HashSet<Person> bubbleAsHashset = new HashSet<Person>(thisBubble);
+			for (Person p: thisBubble) {
+				bubbleContainsEveryoneInWorkplace = p.getWorkBubble().containsAll(bubbleAsHashset);
+			}
+		}
+		Assert.assertTrue(bubbleContainsEveryoneInWorkplace);
+
+	}
+	@Test
+	public void checkPeopleGoToTheirWorkplace() {
+		// check the movement of the population to their workplaces
+		WorldBankCovid19Sim sim = helperFunctions.CreateDummySim("src/main/resources/workplace_bubbles_params.txt");
+		makePeopleLeaveTheHouseEachDay(sim);
+		// make everyone decide to go to their workplace
+		sim.params.prob_go_to_work = 1.1d;
+		sim.start();
+		// run for three ticks (people leave the house at tick 2 and leave work at tick 4)
+		int numTicks = 3;
+		helperFunctions.runSimulationForTicks(sim, numTicks);
+		// determine if everyone has travelled to their workplace
+		// some jobs are based in the community, change to match
+		boolean allAtWork = true;
+		for (Person p: sim.agents) {
+			if (!(p.getLocation() instanceof Workplace) && !p.isUnemployed() && !p.visitingNow()) {
```

Comment:
chat about performance of java parsing - && and || assess left to right, so we can cut processing corners by putting !p.isUnemployed() first